src/sage/arith/multi_modular.pyx

```diff
@@ -1,3 +1,4 @@
+# sage.doctest: optional - primecountpy
```

Comment:
It would be so cool if all files could be like this one with only the declaration of the overall optional dependencies. Of course it's much more complicated than that in practice as some optional tags concern only a few tests in a file and are really iff some optional package (e.g., igraph) has been installed. But for dependencies that are repeated on almost all lines, it could ease our life to force the entire file to depend on these packages.

Reply:
Yes, and in some cases we can achieve this by splitting out parts of files to a separate file (but probably we should do this only where it seems natural).
- For example, with @kwankyu we did this in #35230 - to avoid having to mark one half of each of several files by `# optional - sage.libs.singular`.
- See also https://github.com/sagemath/sage/pull/35742#discussion_r1225466661